yt_dlp/extractor/atresplayer.py

```diff
@@ -92,11 +98,11 @@ def _real_extract(self, url):
                 continue
             src_type = source.get('type')
             if src_type == 'application/vnd.apple.mpegurl':
-                formats, subtitles = self._extract_m3u8_formats(
+                formats, subtitles = self._extract_m3u8_formats_and_subtitles(
                     src, video_id, 'mp4', 'm3u8_native',
                     m3u8_id='hls', fatal=False)
             elif src_type == 'application/dash+xml':
-                formats, subtitles = self._extract_mpd_formats(
+                formats, subtitles = self._extract_mpd_formats_and_subtitles(
                     src, video_id, mpd_id='dash', fatal=False)
```

Comment:
If there can be multiple sources, this discards the formats from all but the last one

Reply:
Mmm, you are right. This code isn't completely mine. I overlooked this. The list of formats can be extended but, what if there are two sources of subtitles for the same language? For example, normal subs and subs tailored for deaf people. The dictionaries could be merged somehow, but it could break language ids.

Reply:
there is a `self._merge_subtitles` function for this

Reply:
Oh, thanks! And thanks for the review, too!